File: app/src/main/java/com/nptel/courses/online/ui/main/MainViewModel.java
package com.nptel.courses.online.ui.main;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.nptel.courses.online.entities.Course;

public class MainViewModel extends ViewModel {

    private MutableLiveData<Course> selectedCourseLiveData;

    public MutableLiveData<Course> getSelectedCourse() {
        if (selectedCourseLiveData == null)
            selectedCourseLiveData = new MutableLiveData<>();
        return selectedCourseLiveData;
    }

    public void setSelectedCourse(Course selectedCourse) {
        selectedCourseLiveData.setValue(selectedCourse);
    }
}

File: app/src/main/java/com/nptel/courses/online/ui/main/MainViewModel.kt
package com.nptel.courses.online.ui.main

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.nptel.courses.online.entities.Course
import com.nptel.courses.online.entities.Playlist
import com.nptel.courses.online.retrofit.RetrofitServices
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.launch
import javax.inject.Inject

@HiltViewModel
class MainViewModel @Inject constructor(
    var retrofitServices: RetrofitServices,
    var selectedCourse: Course
) : ViewModel() {

    private val _status = MutableLiveData<Int>()
    val status: LiveData<Int> = _status

    private val _playlists = MutableLiveData<List<Playlist>>()
    val playlists: LiveData<List<Playlist>> = _playlists

    fun getPlaylists() {
        viewModelScope.launch {
            try {
                val playlists = retrofitServices.getPlaylists(selectedCourse.id)
                _status.value = 1
                _playlists.value = playlists
            } catch (throwable: Throwable) {
                _status.value = 1
                _playlists.value = null
            }
        }
    }

    init {
        getPlaylists()
    }
}